### PR TITLE
Docs: Document Custom Fastify Server configuration settings

### PR DESCRIPTION
**Pull request by dthyresson**
Fixes: https://github.com/redwoodjs/redwoodjs.com/issues/926

Document how to customize the Fastify sever configuration with the new `severConfig` api redwood.toml option.

### CODE REVIEW

File: docs/appConfiguration.md [hunk 1]
```diff
@@ -202,16 +252,17 @@ Let's look at an example:
 [web]
   title = "App running on ${APP_TITLE}"
   port = "${PORT:8910}"
-  apiUrl = "${API_URL:/.redwood/functions}"
+  apiUrl = "${API_URL}/.redwood/functions"
```
Comment (dac09) [inline docs/appConfiguration.md:255]: Think this change isn't correct! The one before was correct :)

Comment (dthyresson) [inline docs/appConfiguration.md:255]:
```suggestion
  apiUrl = "${API_URL:/.redwood/functions}"
```

File: docs/appConfiguration.md [hunk 2]
```diff
@@ -125,19 +125,69 @@ API_KEY=...
...
-| `host`   | Hostname to listen on   | `'localhost'` | `development` |
-| `port`   | Port to listen on       | `8911`        | `development` |
-| `path`   | Path to the api side    | `'./api'`     | `both`        |
-| `target` | Target for the api side | `'node'`      | `both`        |
+| Key            | Description                         | Default       | Context       |
+| :------------- | :---------------------------------- | :------------ | :------------ |
+| `host`         | Hostname to listen on               | `'localhost'` | `development` |
+| `port`         | Port to listen on                   | `8911`        | `development` |
+| `path`         | Path to the api side                | `'./api'`     | `both`        |
+| `serverConfig` | Path to the `server.config.js` file | `'./api'`     | `both`        |
```
Comment (dac09) [inline docs/appConfiguration.md:133]: I think the default is `./api/server.config.js`

File: docs/appConfiguration.md [hunk 3]
```diff
@@ -125,19 +125,69 @@ API_KEY=...
...
+
+For the [Fastify Server options](https://www.fastify.io/docs/latest/Reference/Server/#factory) that you can set, see: https://www.fastify.io/docs/latest/Reference/Server/#factory.
+
+Examples include: logger settings, timeouts, maximum payload limits, and more.
+
+> **Note:** This configuration does not apply in a serverless deploy.
+
+```js
+// server.config.js
+/**
```
Comment (dac09) [inline docs/appConfiguration.md:152]: Do you think we could remove this comment block? Repeated above I believe

File: docs/appConfiguration.md [hunk 4]
```diff
@@ -125,19 +125,69 @@ API_KEY=...
...
+```
+
+You may elect to configure different `server.config.js` based on your deployment environment and take advantage of [# Using Environment Variables in redwood.toml](#using-environment-variables-in-redwoodtoml).
+
+Given an environment variable `DEPLOY_ENVIRONMENT` that declares `developmnent`, `staging`, `production`:
+
+```
+[api]
+  port = 8911
+  serverConfig = = "./api/${DEPLOY_ENVIRONMENT}-server.config.s"
```
Comment (dac09) [inline docs/appConfiguration.md:182]:
```suggestion
  serverConfig = = "./api/${DEPLOY_ENVIRONMENT}-server.config.js"
```

**Review by dac09 — CHANGES_REQUESTED** [review]

**Comment by dac09** [comment]
@dthyresson mind resolving the conflicts?

**Review by dac09 — APPROVED** [review]

File: docs/appConfiguration.md (hunk 3 above)
Comment (dac09) [reply docs/appConfiguration.md:152]: I think you may have resolved this accidentally (im ok if you don't want to remove the comments too!)

**Pull request merged**